src/probnum/diffeq/odefiltsmooth/ivpfiltsmooth.py

```diff
@@ -105,16 +113,8 @@ def _odesmooth(self, ode_solution, **kwargs):
         -------
         smoothed_solution: ODESolution
         """
-        ivp_filter_posterior = ode_solution._kalman_posterior
-        ivp_smoother_posterior = self.gfilt.smooth(ivp_filter_posterior)
-
-        smoothed_solution = ODESolution(
-            times=ivp_smoother_posterior.locations,
-            rvs=ivp_smoother_posterior.state_rvs,
-            solver=ode_solution._solver,
-        )
-
-        return smoothed_solution
+        smoothing_posterior = self.gfilt.smooth(ode_solution.kalman_posterior)
+        return KalmanODESolution(smoothing_posterior)
```

Comment:
since sol.kalman_posterior is a public attribute, one could just change this as an attribute, i.e.

`ode_solution.kalman_posterior = self.gfilt.smooth(ode_solution.kalman_posterior); return ode_solution`. Which option do you prefer?

Reply:
The version you implemented seems cleaner to me :) But I don't have a strong preferenece so feel free to change it if you want to.